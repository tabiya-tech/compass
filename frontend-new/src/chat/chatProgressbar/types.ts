export enum ConversationPhase {
  INITIALIZING = "INITIALIZING",
  INTRO = "INTRO",
  COLLECT_EXPERIENCES = "COLLECT_EXPERIENCES",
  DIVE_IN = "DIVE_IN",
  ENDED = "ENDED",
  UNKNOWN = "UNKNOWN"
}

export type CurrentPhase = {
  percentage: number
  phase: ConversationPhase
}

export const defaultCurrentPhase = {
  phase: ConversationPhase.INITIALIZING,
  percentage: 0
}
