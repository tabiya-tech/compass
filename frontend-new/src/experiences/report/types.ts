import { Experience } from "src/experiences/experiencesDrawer/experienceService/experiences.types";

export type ReportProps = {
  name: string;
  email: string;
  phone: string;
  address: string;
  experiences: Experience[];
  conversationConductedAt: string | null;
};

export enum ExportFormat {
  PDF = "pdf",
  DOCX = "docx",
}

export interface IReportFormatProvider {
  download: () => void;
}
