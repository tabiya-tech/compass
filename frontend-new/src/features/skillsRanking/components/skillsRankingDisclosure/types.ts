export const jobSeekerComparisonLabels = ["LOWEST", "SECOND LOWEST", "MIDDLE", "SECOND HIGHEST", "HIGHEST"]