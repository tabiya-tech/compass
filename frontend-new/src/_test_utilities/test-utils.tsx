// Based on https://testing-library.com/docs/react-testing-library/setup/
import React, { ReactElement, ReactNode } from "react";
import { render, renderHook, RenderHookOptions, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";

// Import the Firebase mock utilities
import "src/_test_utilities/firebaseMock";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/isOnlineProvider/IsOnlineProvider";
jest.mock("firebase/compat/app", () => {
  return {
    initializeApp: jest.fn(),
    auth: jest.fn().mockReturnValue({
      signOut: jest.fn(),
      currentUser: {
        getIdToken: jest.fn(),
        getTokenResult: jest.fn(),
      },
      onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
      signInWithPopup: jest.fn(),
      GoogleAuthProvider: { PROVIDER_ID: "google.com" },
    }),
  };
});

jest.mock("firebaseui", () => {
  return {
    auth: {
      AuthUI: {
        start: jest.fn(),
        getInstance: jest.fn().mockReturnValue({
          start: jest.fn(),
          reset: jest.fn(),
        }),
        reset: jest.fn(),
      },
    },
  };
});

jest.mock("@sentry/react", () => ({
  withSentry: (Component: any) => Component,
  getFeedback: jest.fn(),
  withProfiler: (Component: any) => Component,
  wrapCreateBrowserRouter: (Component: any) => Component,
}));

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<T>(
  hook: () => T,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { wrapper?: React.ComponentType }
) {
  const CombinedWrapper = ({ children }: { children: ReactNode }) => {
    const Wrapper = options?.wrapper ?? React.Fragment;
    return (
      <AllTheProviders>
        <Wrapper>{children}</Wrapper>
      </AllTheProviders>
    );
  };

  return renderHook(hook, { wrapper: CombinedWrapper, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
