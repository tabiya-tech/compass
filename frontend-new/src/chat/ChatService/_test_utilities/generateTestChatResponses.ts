import { ConversationMessage, ConversationMessageSender } from "src/chat/ChatService/ChatService.types";

export const generateTestChatResponses = (): ConversationMessage[] => {
  return [
    {
      message: "Hello! How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "I can help you with that. What is your name?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
  ];
};

export const generateTestHistory = (): ConversationMessage[] => {
  return [
    {
      message: "Hello! How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "I can help you with that. What is your name?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "My name is John.",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.USER,
    },
    {
      message: "Nice to meet you, John. How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
  ];
};
