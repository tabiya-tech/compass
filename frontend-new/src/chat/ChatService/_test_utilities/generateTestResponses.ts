import { LastMessage, RootObject } from "src/chat/ChatService/ChatService.types";

export const generateRootObjectResponse = (): RootObject => ({
  last: {
    message_for_user: "Hello",
    finished: false,
    agent_type: "WelcomeAgent",
    reasoning: "Initial greeting",
    agent_response_time_in_sec: 1.0,
    llm_stats: [
      {
        error: "",
        prompt_token_count: 50,
        response_token_count: 10,
        response_time_in_sec: 1.0,
      },
    ],
  },
  conversation_context: {
    all_history: {
      turns: [
        {
          index: 1,
          input: {
            message: "Hello",
          },
          output: {
            message_for_user: "Hi there!",
            finished: false,
            agent_type: "WelcomeAgent",
            reasoning: "Greeting response",
            agent_response_time_in_sec: 0.5,
            llm_stats: [
              {
                error: "",
                prompt_token_count: 30,
                response_token_count: 10,
                response_time_in_sec: 0.5,
              },
            ],
          },
        },
      ],
    },
    history: {
      turns: [],
    },
    summary: "",
  },
});

export const generateLastMessageResponse = (): LastMessage => ({
  message_for_user: "Hi there!",
  finished: false,
  agent_type: "WelcomeAgent",
  reasoning: "Greeting response",
  agent_response_time_in_sec: 1.0,
  llm_stats: [
    {
      error: "",
      prompt_token_count: 30,
      response_token_count: 10,
      response_time_in_sec: 0.5,
    },
  ],
});
