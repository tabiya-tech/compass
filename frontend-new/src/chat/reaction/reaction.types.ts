import { ReactionError } from "src/error/commonErrors";

export enum ReactionKind {
  LIKED = "LIKED",
  DISLIKED = "DISLIKED",
}

export enum DislikeReason {
  INAPPROPRIATE_TONE = "INAPPROPRIATE_TONE",
  OFFENSIVE_LANGUAGE = "OFFENSIVE_LANGUAGE",
  BIASED = "BIASED",
  INCORRECT_INFORMATION = "INCORRECT_INFORMATION",
  IRRELEVANT = "IRRELEVANT",
  CONFUSING = "CONFUSING",
}

export const DislikeReasonMessages = {
  [DislikeReason.INAPPROPRIATE_TONE]: "Inappropriate Tone",
  [DislikeReason.OFFENSIVE_LANGUAGE]: "Offensive Language",
  [DislikeReason.BIASED]: "Biased",
  [DislikeReason.INCORRECT_INFORMATION]: "Incorrect Information",
  [DislikeReason.IRRELEVANT]: "Irrelevant",
  [DislikeReason.CONFUSING]: "Confusing",
};

// Base abstract class for reactions
export abstract class BaseReaction {
  abstract readonly kind: ReactionKind;
}

// Like reaction class - no additional properties needed
export class LikeReaction extends BaseReaction {
  readonly kind = ReactionKind.LIKED;
  readonly reasons: DislikeReason[] = [];

  constructor() {
    super();
    if(this.reasons.length > 0){
      throw new ReactionError("Like Reactions should not have any reasons");
    }
  }
}

// Dislike reaction class - includes reasons
export class DislikeReaction extends BaseReaction {
  readonly kind = ReactionKind.DISLIKED;
  
  constructor(public readonly reasons: DislikeReason[]) {
    super();
    if(reasons.length === 0){
      throw new ReactionError("Dislike Reactions should always have a reason");
    }
  }
}