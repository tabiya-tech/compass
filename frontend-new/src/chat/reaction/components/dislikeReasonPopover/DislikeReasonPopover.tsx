import React, { useContext } from "react";
import { useTranslation } from "react-i18next";
import { Box, Popover, Typography, useTheme } from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";
import PrimaryIconButton from "src/theme/PrimaryIconButton/PrimaryIconButton";
import PrimaryButton from "src/theme/PrimaryButton/PrimaryButton";
import { DislikeReason, DislikeReasonMessages } from "src/chat/reaction/reaction.types";
import { IsOnlineContext } from "src/app/isOnlineProvider/IsOnlineProvider";

export interface DislikeReasonPopoverProps {
  anchorEl: HTMLElement | null; // It allows the popover to know which element to attach to and appear next to.
  open: boolean;
  onClose: (reasons: DislikeReason[]) => void;
}

const uniqueId = "9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e";

export const DATA_TEST_ID = {
  POPOVER: `dislike-reason-popover-${uniqueId}`,
  CONTAINER: `dislike-reason-popover-container-${uniqueId}`,
  TITLE: `dislike-reason-popover-title-${uniqueId}`,
  BUTTON: `dislike-reason-popover-button-${uniqueId}`,
  CLOSE_ICON: `dislike-reason-popover-close-button-${uniqueId}`,
  CLOSE_ICON_BUTTON: `dislike-reason-popover-close-icon-button-${uniqueId}`,
};

export const DislikeReasonPopover: React.FC<DislikeReasonPopoverProps> = ({
  anchorEl,
  open,
  onClose,
}) => {
  const theme = useTheme();
  const { t } = useTranslation();
  const isOnline = useContext(IsOnlineContext)

  const handleReasonClick = (reason: DislikeReason) => {
    // for now, our ui only allows selecting one reason
    onClose([reason]);
  };

  return (
    <Popover
      open={open}
      anchorEl={anchorEl}
      anchorOrigin={{
        vertical: "bottom",
        horizontal: "center",
      }}
      transformOrigin={{
        vertical: "top",
        horizontal: "center",
      }}
      onClose={() => onClose([])}
      data-testid={DATA_TEST_ID.POPOVER}
    >
      <Box
        display="flex"
        flexDirection="column"
        maxWidth={400}
        gap={theme.fixedSpacing(theme.tabiyaSpacing.md)}
        sx={{ padding: theme.fixedSpacing(theme.tabiyaSpacing.lg) }}
        data-testid={DATA_TEST_ID.CONTAINER}
      >
        <Box display="flex" justifyContent="space-between" alignItems="center">
          <Typography variant="subtitle1" color={theme.palette.text.secondary} data-testid={DATA_TEST_ID.TITLE}>
            {t("tell_us_what_the_issue_is")}
          </Typography>
          <PrimaryIconButton
            onClick={() => onClose([])} // close without selecting a reason
            title="close feedback"
            sx={{ color: theme.palette.text.secondary }}
            data-testid={DATA_TEST_ID.CLOSE_ICON_BUTTON}
          >
            <CloseIcon data-testid={DATA_TEST_ID.CLOSE_ICON} />
          </PrimaryIconButton>
        </Box>
        <Box display="flex" flexDirection="row" flexWrap="wrap" gap={theme.fixedSpacing(theme.tabiyaSpacing.sm)}>
          {Object.entries(DislikeReasonMessages).map(([enumValue, message]) => (
            <PrimaryButton
              key={enumValue}
              onClick={() => handleReasonClick(enumValue as DislikeReason)}
              title={t(enumValue.toLowerCase())}
              style={{ color: theme.palette.text.secondary }}
              data-testid={DATA_TEST_ID.BUTTON}
              disabled={!isOnline}
            >
              {t(enumValue.toLowerCase())}
            </PrimaryButton>
          ))}
        </Box>
      </Box>
    </Popover>
  );
};

export default DislikeReasonPopover;
