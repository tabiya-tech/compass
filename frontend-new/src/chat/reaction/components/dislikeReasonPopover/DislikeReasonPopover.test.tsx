// mute the console
import "src/_test_utilities/consoleMock";

import React from "react";
import { render, screen, userEvent} from "src/_test_utilities/test-utils";
import DislikeReasonPopover, {
  DATA_TEST_ID,
} from "src/chat/reaction/components/dislikeReasonPopover/DislikeReasonPopover";
import { DislikeReason, DislikeReasonMessages } from "src/chat/reaction/reaction.types";
import { mockBrowserIsOnLine } from "src/_test_utilities/mockBrowserIsOnline";

describe("ReactionReasonPopover", () => {

  beforeEach(() => {
    jest.clearAllMocks();
    mockBrowserIsOnLine(true);
  });

  describe("render tests", () => {
    test("should render the reaction reason popover correctly", () => {
      // GIVEN an anchor element
      const anchorEl = document.createElement("div");

      // WHEN the component is rendered
      render(<DislikeReasonPopover anchorEl={anchorEl} open={true} onClose={jest.fn()} />);

      // THEN expect the container to be in the document
      const container = screen.getByTestId(DATA_TEST_ID.CONTAINER);
      expect(container).toBeInTheDocument();
      // AND the title to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.TITLE)).toBeInTheDocument();
      // AND the close icon Button to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.CLOSE_ICON_BUTTON)).toBeInTheDocument();
      // AND the close icon to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.CLOSE_ICON)).toBeInTheDocument();
      // AND expect all reason buttons to be rendered
      const reasonButtons = screen.getAllByTestId(DATA_TEST_ID.BUTTON);
      expect(reasonButtons).toHaveLength(Object.keys(DislikeReasonMessages).length);
      // AND the buttons to have the correct text content
      const reasons = Object.values(DislikeReasonMessages);
      reasons.forEach((reason, index) => {
        expect(reasonButtons[index]).toHaveTextContent(reason);
        expect(reasonButtons[index]).toBeEnabled();
      });
      // AND expect the popover to be anchored to the correct element.
      expect(screen.getByTestId(DATA_TEST_ID.POPOVER)).toBeInTheDocument();
      // AND expect the component to match snapshot
      expect(container).toMatchSnapshot();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should render the popover with disabled buttons when the browser is offline", () => {
      // GIVEN an anchor element
      const anchorEl = document.createElement("div");

      // AND the browser is offline
      mockBrowserIsOnLine(false);

      // WHEN the component is rendered
      render(<DislikeReasonPopover anchorEl={anchorEl} open={true} onClose={jest.fn()} />);

      // THEN expect the container to be in the document
      const container = screen.getByTestId(DATA_TEST_ID.CONTAINER);
      expect(container).toBeInTheDocument();
      // AND the title to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.TITLE)).toBeInTheDocument();
      // AND the close icon Button to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.CLOSE_ICON_BUTTON)).toBeInTheDocument();
      // AND the close icon to be in the document
      expect(screen.getByTestId(DATA_TEST_ID.CLOSE_ICON)).toBeInTheDocument();
      // AND expect all reason buttons to be rendered
      const reasonButtons = screen.getAllByTestId(DATA_TEST_ID.BUTTON);
      expect(reasonButtons).toHaveLength(Object.keys(DislikeReasonMessages).length);
      // AND the buttons to have the correct text content
      const reasons = Object.values(DislikeReasonMessages);
      reasons.forEach((reason, index) => {
        expect(reasonButtons[index]).toHaveTextContent(reason);
        expect(reasonButtons[index]).toBeDisabled();
      });
      // AND expect the popover to be anchored to the correct element.
      expect(screen.getByTestId(DATA_TEST_ID.POPOVER)).toBeInTheDocument();
      // AND expect the component to match snapshot
      expect(container).toMatchSnapshot();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    })
  });

  describe("action tests", () => {
    test("should call onClose with a reason when a reason is selected", async () => {
      // GIVEN the popover is rendered and open
      const givenOnClose = jest.fn();
      const givenComponent = (
        <DislikeReasonPopover
          anchorEl={document.createElement("div")}
          open={true}
          onClose={givenOnClose}
        />
      );
      render(givenComponent);

      // WHEN a reason button is clicked
      const reasonButton = screen.getAllByTestId(DATA_TEST_ID.BUTTON)[0];
      await userEvent.click(reasonButton);

      // THEN expect the onClose callback to be called with the correct reason
      expect(givenOnClose).toHaveBeenCalledWith([DislikeReason.INAPPROPRIATE_TONE]);
      // AND expect onClose to be called
      expect(givenOnClose).toHaveBeenCalled();
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should call onClose with an empty array when clicking the close icon", async () => {
      // GIVEN the popover is rendered and open
      const givenOnClose = jest.fn();
      render(
        <DislikeReasonPopover
          anchorEl={document.createElement("div")}
          open={true}
          onClose={givenOnClose}
        />
      );

      // WHEN the close icon button is clicked
      const closeIconButton = screen.getByTestId(DATA_TEST_ID.CLOSE_ICON_BUTTON);
      await userEvent.click(closeIconButton);

      // THEN expect onClose to be called
      expect(givenOnClose).toHaveBeenCalledWith([]);
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should call onClose with an empty array when clicking outside the popover", async () => {
      // GIVEN the popover is rendered and open
      const givenOnClose = jest.fn();
      render(
        <DislikeReasonPopover
          anchorEl={document.createElement("div")}
          open={true}
          onClose={givenOnClose}
        />
      );

      // WHEN clicking outside the popover
      // eslint-disable-next-line testing-library/no-node-access
      const backdrop = document.querySelector(".MuiBackdrop-root");
      if (backdrop) {
        await userEvent.click(backdrop);
      }

      // THEN expect onClose to be called
      expect(givenOnClose).toHaveBeenCalledWith([]);
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });

    test("should call onClose with an empty array when pressing escape", async () => {
      // GIVEN the popover is rendered and open
      const givenOnClose = jest.fn();
      render(
        <DislikeReasonPopover
          anchorEl={document.createElement("div")}
          open={true}
          onClose={givenOnClose}
        />
      );

      // WHEN the escape key is pressed
      await userEvent.keyboard("{Escape}");

      // THEN expect onClose to be called
      expect(givenOnClose).toHaveBeenCalledWith([]);
      // AND expect no errors or warnings to have occurred
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
    });
  });
});
