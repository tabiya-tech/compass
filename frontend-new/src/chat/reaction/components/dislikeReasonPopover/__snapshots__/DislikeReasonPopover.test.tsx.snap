// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionReasonPopover render tests should render the popover with disabled buttons when the browser is offline 1`] = `
<div
  class="MuiBox-root css-vsmmf2"
  data-testid="dislike-reason-popover-container-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
>
  <div
    class="MuiBox-root css-69i1ev"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-b9u22t-MuiTypography-root"
      data-testid="dislike-reason-popover-title-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
    >
      Please tell us what the issue is?
    </h6>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1soxa02-MuiButtonBase-root-MuiIconButton-root"
      data-testid="dislike-reason-popover-close-icon-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      tabindex="0"
      title="close feedback"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="dislike-reason-popover-close-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-17outbb"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Inappropriate Tone"
      type="button"
    >
      Inappropriate Tone
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Offensive Language"
      type="button"
    >
      Offensive Language
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Biased"
      type="button"
    >
      Biased
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Incorrect Information"
      type="button"
    >
      Incorrect Information
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Irrelevant"
      type="button"
    >
      Irrelevant
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      disabled=""
      style="color: rgb(65, 64, 61);"
      tabindex="-1"
      title="Confusing"
      type="button"
    >
      Confusing
    </button>
  </div>
</div>
`;

exports[`ReactionReasonPopover render tests should render the reaction reason popover correctly 1`] = `
<div
  class="MuiBox-root css-vsmmf2"
  data-testid="dislike-reason-popover-container-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
>
  <div
    class="MuiBox-root css-69i1ev"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-b9u22t-MuiTypography-root"
      data-testid="dislike-reason-popover-title-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
    >
      Please tell us what the issue is?
    </h6>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1soxa02-MuiButtonBase-root-MuiIconButton-root"
      data-testid="dislike-reason-popover-close-icon-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      tabindex="0"
      title="close feedback"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="dislike-reason-popover-close-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-17outbb"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Inappropriate Tone"
      type="button"
    >
      Inappropriate Tone
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Offensive Language"
      type="button"
    >
      Offensive Language
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Biased"
      type="button"
    >
      Biased
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Incorrect Information"
      type="button"
    >
      Incorrect Information
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Irrelevant"
      type="button"
    >
      Irrelevant
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-fph7a5-MuiButtonBase-root-MuiButton-root"
      data-testid="dislike-reason-popover-button-9f3e2d1c-8b7a-4c6d-a5e9-2f1d8c7b3a4e"
      style="color: rgb(65, 64, 61);"
      tabindex="0"
      title="Confusing"
      type="button"
    >
      Confusing
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
